extjsdk/src/main/java/io/vantiq/extjsdk/ExtensionWebSocketListener.java

```diff
@@ -314,6 +316,16 @@ public void onMessage(ResponseBody body) {
         else if (msg.get("op") == null) {
             log.debug("Http response received");
             if (client.isAuthed()) {
+                // Is an error message before successful connection to the target source
+                // This is most likely a failure related to a source connection request
+                if (!client.isConnected() && (Integer) msg.get("status") >= 300) {
+                    if (msg.get("body") instanceof Map && ((Map)msg.get("body")).get("code") instanceof String) {
+                        String errorCode = (String) ((Map)msg.get("body")).get("code");
+                        if (errorCode.startsWith(CONNECT_ERROR)) {
```

Comment:
Seems like if we get a different error (don't know, but things change over time), this will still hang.  Seems like any error code should abort the same way.

Reply:
Fixed but not committed/pushed yet.